### DISCUSSION TITLE
Saving Form Group with a relationship and unique text constraint

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#11111] [Help]
### Package

Form builder

### Package Version

v3.2.14

### How can we help you?

Hello,

I have added a group of fields with a relationship that collects some data to create unique records based on the input of that group.  My problem is that the form tries to create a new record with the same values instead of finding the matching record that already exists.

```
Group::make()
                    ->relationship('source')
                    ->schema([
                        Forms\Components\Select::make('sourcetype_id')
                            ->label('Source Type')
                            ->reactive()
                            ->options(
                                function (\Filament\Forms\Get $get) {
                                    return (SourceType::all()->pluck('name', 'id'));
                                }
                            )
                            ->required(),
                        Forms\Components\TextInput::make('data')
                            ->label('Source')
                            ->required()
                            ->reactive()
                            ->maxLength(255),
                    ]),
```

When I try to save a record it returns a unique error.
```
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'mydata-1' for key 'sources.sources_data_sourcetype_id_unique'
```

The relationship has a unique constraint on the 'data'.
```
$table->unique(['data', 'sourcetype_id']);
```

So how would I use the two fields to find an existing record if it exists and use that instead of creating a new record every time?

Additionally the grouped fields do not seem to load data on the edit page.  Is this just a limitation of using a group of fields with a relationship?

**Comment by medfrk** [comment]
try to add the constraint   
 ->unique(ignoreRecord: true)
 
like this:
Forms\Components\Select::make('sourcetype_id')
                            ->label('Source Type')
                            ->reactive()
                            ->options(
                                function (\Filament\Forms\Get $get) {
                                    return (SourceType::all()->pluck('name', 'id'));
                                }
                            )
                            ->unique(ignoreRecord: true)
                            ->required(),
Hope that's help you

**Reply by yaplej** [reply]
So that just results in a message that the text has already been used.  It does not link to that existing record.

Same as this:
```
->unique(ignorable: fn ($record) => $record)
```
